csharp/ql/src/experimental/dataflow/flowsources/AuthCookie.qll

```diff
@@ -0,0 +1,192 @@
+/**
+ * Provides classes and predicates for detecting insecure cookies.
+ */
+
+import csharp
+import semmle.code.csharp.frameworks.microsoft.AspNetCore
+
+/**
+ * Holds if the expression is a variable with a sensitive name.
+ */
+predicate isCookieWithSensitiveName(Expr cookieExpr) {
+  exists(AuthCookieNameConfiguration dataflow, DataFlow::Node source, DataFlow::Node sink |
+    dataflow.hasFlow(source, sink) and
+    sink.asExpr() = cookieExpr
+  )
+}
+
+/**
+ * Tracks if a variable with a sensitive name is used as an argument.
+ */
+private class AuthCookieNameConfiguration extends DataFlow::Configuration {
+  AuthCookieNameConfiguration() { this = "AuthCookieNameConfiguration" }
+
+  private predicate isAuthVariable(Expr expr) {
+    exists(string val |
+      (
+        val = expr.getValue() or
```

Comment:
Why this line?

Reply:
The logic is to check variable/member name and its value. Like:
```cs
var cookieName = "session";
```